src/ObjCRuntime/Runtime.CoreCLR.cs

```diff
@@ -233,6 +233,15 @@ static Type GetMethodReturnType (MethodBase method)
 			return null;
 		}
 
+		// This should work like mono_class_from_mono_type.
+		static unsafe MonoObject* TypeToClass (MonoObject* typeobj)
+		{
+			var type = (Type) GetMonoObjectTarget (typeobj);
+			if (type.IsByRef)
```

Comment:
For dereferencing, are there ever more than one level of referencing?

Reply:
No, there's only ever one level of referencing in this case.